src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -564,29 +564,28 @@ public IReadOnlyList<LoggerClass> GetLogClasses(IEnumerable<ClassDeclarationSynt
 
             private (string? loggerField, bool multipleLoggerFields) FindLoggerField(SemanticModel sm, TypeDeclarationSyntax classDec, ITypeSymbol loggerSymbol)
             {
+                List<IFieldSymbol> fields = new();
+
+                INamedTypeSymbol? classType = sm.GetDeclaredSymbol(classDec, _cancellationToken);
+                while (classType is { } ct && ct.SpecialType != SpecialType.System_Object)
+                {
+                    fields.AddRange(classType.GetMembers().OfType<IFieldSymbol>());
```

Comment:
💡 Should `private` fields of base types be excluded?

Reply:
Good question. I think at the moment, it's reasonable what it's doing: it's generating the code for a field that's not accessible, resulting in `error CS0122: 'BaseClass._logger' is inaccessible due to its protection level`.  I'd be equally happy to see that as I would a custom code analysis error.  I'll delegate the ultimate decision to someone else though! :)

Reply:
I'd be more worried about having a duplicate type in the base (which isn't a duplicate after eliminating private members).

Reply:
Multiple declarations of any `ILogger` fields result in `error SYSLIB1020: Found multiple fields of type Microsoft.Extensions.Logging.ILogger in class DerivedClass`

Reply:
@sharwell - any further thoughts on this one?

Reply:
I think we should exclude `private` fields of base types. Imagine this scenario:

```C#
public class BaseClass
{
    private ILogger _logger;

    public BaseClass(ILogger logger) => _logger = logger;
}

public partial class DerivedClass : BaseClass
{
    private ILogger _logger;

    public DerivedClass(ILogger logger) : base(logger)  => _logger = logger;

    [LoggerMessage(0, LogLevel.Debug, "Test.")]
    public partial void Test();
}
```

With the current code you'd get:
`error SYSLIB1020: Found multiple fields of type Microsoft.Extensions.Logging.ILogger in class DerivedClass`

which doesn't seem like a great experience.

Reply:
Oh yes, that's not good!